homeassistant/components/tradfri/fan.py

```diff
@@ -17,7 +17,24 @@
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .base_class import TradfriBaseDevice
-from .const import ATTR_AUTO, CONF_GATEWAY_ID, DEVICES, DOMAIN, KEY_API
+from .const import (
+    ATTR_AUTO,
+    ATTR_MAX_FAN_STEPS,
+    CONF_GATEWAY_ID,
+    DEVICES,
+    DOMAIN,
+    KEY_API,
+)
+
+
+def _from_fan_percentage(percentage: int) -> int:
+    """Convert percent to a value that the Tradfri API understands."""
+    return round(percentage / 100 * ATTR_MAX_FAN_STEPS)
```

Comment:
The allowed fan speed is between 2 and 50. This function will generate a alue between 0 and 49.

Reply:
The calculation in the other helper function was quite possibly suffering from the same calculation error. Made some changes and added a few test cases.

Reply:
Hm, just realised I'm not covering the case of 0. Will update.